Open Judge System/Workers/OJS.Workers.LocalWorker/LocalWorkerService.cs

```diff
@@ -60,53 +49,59 @@ protected override void OnStop()
         {
             logger.Info("LocalWorkerService stopping...");
 
-            foreach (var job in this.jobs)
-            {
-                job.Stop();
-                logger.InfoFormat("{0} stopped.", job.Name);
-            }
+            this.StopJobs(this.jobs);
 
             Thread.Sleep(10000);
 
-            foreach (var thread in this.threads)
+            this.StopThreads(this.threads);
+
+            logger.Info("LocalWorkerService stopped.");
+        }
+
+        private void PopulateJobsAndThreads(
+            ICollection<IJob> jobsToPopulate,
+            ICollection<Thread> threadsToPopulate,
+            ConcurrentQueue<int> submissionsForProcessing)
+        {
+            for (var i = 1; i <= Settings.ThreadsCount; i++)
             {
-                thread.Abort();
-                logger.InfoFormat("{0} aborted.", thread.Name);
+                var job = new SubmissionJob(
+                    $"Job №{i}",
+                    submissionsForProcessing);
+
+                var thread = new Thread(job.Start) { Name = $"Thread №{i}" };
+
+                jobsToPopulate.Add(job);
+                threadsToPopulate.Add(thread);
             }
+        }
 
-            logger.Info("LocalWorkerService stopped.");
+        private void StartThreads(IEnumerable<Thread> threadsToStarts)
+        {
+            foreach (var thread in threadsToStarts)
+            {
+                logger.InfoFormat($"Starting {thread.Name}...");
+                thread.Start();
+                logger.InfoFormat($"{thread.Name} started.");
+                Thread.Sleep(234);
+            }
         }
 
-        /// <summary>
-        /// Sets the Processing property to False for all submissions
-        /// thus ensuring that the worker will process them eventually instead
-        /// of getting stuck in perpetual "Processing..." state
-        /// </summary>
-        private void ResetAllProcessingSubmissions()
+        private void StopJobs(IEnumerable<IJob> jobsToStop)
+        {
+            foreach (var job in jobsToStop)
+            {
+                job.Stop();
+                logger.InfoFormat($"{job.Name} stopped.");
+            }
+        }
+
+        private void StopThreads(IEnumerable<Thread> threadsToStop)
```

Comment:
`threads` instead of 'toStop'

Reply:
It is overriding the private field `threads`